enctests/testframework/main.py

```diff
@@ -303,6 +319,126 @@ def vmaf_compare(source_clip, test_ref, testname):
+            for line in lines[:-1]:
+                if " = " not in line:
+                    continue
+                key, value = line.split(" = ")
+                key = key.strip().replace(" ", "_").lower()
+                if key == "max_error":
+                    value, location = value.split(" @ ")
+                    result['max_location'] = location
+                else:
+                    value = value.strip()
+                result[key] = float(value)
+    enc_meta = get_test_metadata_dict(test_ref)
+    enc_meta['results'].update(result)
+
+def assertresults_compare(source_clip, test_ref, testname, comparisontest_info):
+    """
+    Check the results of the tests against known values (or value ranges).
+    We assume that we have already run some tests, and just want to check that the values are good.
+    
+    source_clip -- The original clip that the new media is based on. (not used)
+    test_ref    -- The generated movie we are testing.(not used)
+    testname    -- The testname we are running.
+    comparisontestinfo -- other parameters that can be used to configure the test.
+
+    testresult - Was the test able to run (Completed = Yes)
+    success - Boolean, was the test a success. 
+    """
+    tests = comparisontest_info.get("tests", [])
+    enc_meta = get_test_metadata_dict(test_ref)
+    result = enc_meta['results']
+    resultstatus = True
+    for test in tests:
+        if "assert" not in test:
+            print("WARNING: no test to run in test:", test, " expecting a field called assert with the test type.")
+            continue
+        testname = test.get("assert")
+        testvalue = test.get("value") # which field to test.
+        if testvalue not in result:
+            print("Skipping test ", test, " since value ", testvalue, " is not in results:", result)
+            continue
+        if testname == "between":
+            if "between" not in test:
+                print("WARNING: Skipping test since there is no between values, in :", test)
+            values = test.get("between")
+
+            resultstatus = result[testvalue] > values[0] and result[testvalue] < values[1]
+        if testname == "greater":
+            if "greater" not in test:
+                print("WARNING: Skipping test since there is no greater values, in :", test)
+            value = test.get("greater")
+            resultstatus = result[testvalue] > value
+        if testname == "less":
+            if "less" not in test:
+                print("WARNING: Skipping test since there is no greater values, in :", test)
+            value = test.get("less")
+            resultstatus = result[testvalue] < value
+ 
+        if testname == "stringmatch":
+            if "string" not in test:
+                print("WARNING: Skipping test since there is no string to match in :", test)
+            value = test.get("string")
+
+            resultstatus = result[testvalue] == value
```

Comment:
Like mentioned above, we should try to use f-strings.
Also, I think there are missing `continue`s in the if tests

Reply:
Should be fixed.